clevertap-core/src/test/java/com/clevertap/android/sdk/AnalyticsManagerTest.kt

```diff
@@ -1,96 +1,335 @@
+                withArg {
+                    JSONAssert.assertEquals(json, it, true)
+                },
+                Constants.NV_EVENT
+            )
+        }
+
+        // setup again, 200 ms has passed
+        every { timeProvider.invoke() } returns 10200
+
+        // Send duplicate PN
+        analyticsManagerSUT.pushNotificationViewedEvent(bundle)
+
+        // verify it was not called again, one time was from before
+        verify(exactly = 1) {
+            eventQueueManager.queueEvent(
+                context,
+                withArg {
+                    JSONAssert.assertEquals(json, it, true)
+                },
+                Constants.NV_EVENT
+            )
+        }
+        confirmVerified(eventQueueManager)
+    }
+
+    @Test
+    fun `clevertap processes PN viewed for same wzrk_id if separated by a span of greater than 2 seconds`() {
+
+        val json = notificationViewedJson(bundle);
+
+        every { timeProvider.invoke() } returns 10000
+
+        // send PN first time
+        analyticsManagerSUT.pushNotificationViewedEvent(bundle)
+
+        verify(exactly = 1) {
+            eventQueueManager.queueEvent(
+                context,
+                withArg {
+                    JSONAssert.assertEquals(json, it, true)
+                },
+                Constants.NV_EVENT
+            )
+        }
+
+        // setup again, 10000 ms has passed
+        every { timeProvider.invoke() } returns 20000
+
+        // Send duplicate PN
+        analyticsManagerSUT.pushNotificationViewedEvent(bundle)
+
+        // verify queue event called again
+        verify(exactly = 2) {
+            eventQueueManager.queueEvent(
+                context,
+                withArg {
+                    JSONAssert.assertEquals(json, it, true)
+                },
+                Constants.NV_EVENT
+            )
+        }
+        confirmVerified(eventQueueManager)
+    }
+
+    @Test
+    fun `clevertap does not process PN Clicked if SDK is set to analytics only`() {
+        cleverTapInstanceConfig.isAnalyticsOnly = true
+
+        // send PN first time
+        analyticsManagerSUT.pushNotificationClickedEvent(bundle)
+
+        verify {
+            eventQueueManager wasNot called
+        }
+        confirmVerified(eventQueueManager)
+    }
+
+    @Test
+    fun `clevertap does not process duplicate (same wzrk_id) PN clicked within 2 seconds - case 2nd click happens in 200ms`() {
+
+        val json = notificationViewedJson(bundle)
+        every { timeProvider.invoke() } returns 10000
+
+        // send PN first time
+        analyticsManagerSUT.pushNotificationClickedEvent(bundle)
+
+        verify(exactly = 1) {
+            eventQueueManager.queueEvent(
+                context,
+                withArg {
+                    JSONAssert.assertEquals(json, it, true)
+                },
+                Constants.RAISED_EVENT
+            )
+        }
+
+        // setup again, 10000 ms has passed
```

Comment:
The difference is 2000ms from the initial mock. Not sure if it should be 200ms based on the test name, but it is not 10000ms